xmpp/auth.py

```diff
@@ -157,7 +154,7 @@ def FeaturesHandler(self,conn,feats):
             node=Node('auth',attrs={'xmlns':NS_SASL,'mechanism':'DIGEST-MD5'})
         elif "PLAIN" in mecs:
             sasl_data='%s\x00%s\x00%s'%(self.username+'@'+self._owner.Server,self.username,self.password)
-            payload = base64.b64encode(sasl_data.encode('utf-8')).decode('utf-8').replace('\r','').replace('\n','')
+            payload = base64.b64encode(sasl_data.encode(CHARSET_ENCODING)).decode(CHARSET_ENCODING).replace('\r','').replace('\n','')
```

Comment:
That can/might/should also be improved?
```suggestion
            payload = ensure_str(base64.b64encode(ensure_binary(sasl_data,CHARSET_ENCODING)),CHARSET_ENCODING).replace('\r','').replace('\n','')
```

Actually, when looking at the previous changes f7149bf8a1f and c7f14d67d0d, I can spot another instance where `six` might want to be employed.

Reply:
With e12316d1fb, the base64 encoding has also been streamlined a bit by introducing another helper function `B64`.